### PR TITLE
NIP-315: User Statuses

### PR DESCRIPTION
**Pull request by jb55**
This NIP enables a way for users to share live statuses such as what music they are listening to, as well as what they are currently doing: work, play, out of office, etc.

[315.md](https://github.com/jb55/nips/blob/user-statuses/315.md)

### CODE REVIEW

**Review by jacany — APPROVED** [review]

**Review by staab — APPROVED** [review]

**Review by vitorpamplona — APPROVED** [review]

**Comment by vitorpamplona** [comment]
> The status MAY include an r, p, e or a tag linking to a URL, profile, note, or parameterized replaceable event.

We might need to keep updating this NIP with any rendering expectations for these links if they aren't included as nostr:bech in the content itself. 

Otherwise, this looks good.

**Review by fiatjaf — APPROVED** [review]

File: 315.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  "content": "Sign up for nostrasia!",
+  "tags": [
+    ["d", "general"],
+    ["r", "https://nostr.world"]
+  ],
+}
+
+{
+  "kind": 30315,
+  "content": "Intergalatic - Beastie Boys",
```
Comment (mattn) [inline 315.md:32]: Does this allow for empty text, text containing line breaks, or long text?

**Comment by jb55** [comment]
> > The status MAY include an r, p, e or a tag linking to a URL, profile, note, or parameterized replaceable event.
> 
> 
> 
> We might need to keep updating this NIP with any rendering expectations for these links (are they always links?) if they aren't included as nostr:bech in the content itself. 
> 
> 
> 
> Otherwise, this looks good. 

I only handle r links atm but figured to add the others for completeness.

File: 315.md (hunk 1 above)
Comment (jb55) [reply 315.md:32]: Its allowed but would be cut off in Damus

Comment (jb55) [reply 315.md:32]: Empty statuses are equivalent to clearing the status in Damus, could make that explicit

File: 315.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  "tags": [
+    ["d", "music"],
+    ["r", "spotify:search:Intergalatic%20-%20Beastie%20Boys"],
+    ["expiration", "1692845589"]
+  ],
+}
+```
+
+Two common status types are defined: `general` and `music`. `general` represent general statuses: "Working", "Hiking", etc.
```
[the remark this answers is not in the document]
Comment (jb55) [reply 315.md:41]: I originally had it this way but then i realized it wouldn't work because the expiry times are different.

**Comment by 0xtrr** [comment]
Implemented support for this in [nostr-tool](https://github.com/0xtrr/nostr-tool/pull/58)

**Review by 0xtrr — APPROVED** [review]

**Comment by fiatjaf** [comment]
584f653ff781fe940a190c61d57ba802e776199e

**Comment by fiatjaf** [comment]
7dc1f02fa70a5b9da7b3867e3faa37f48571d5e1

**Pull request merged**